component-design-manager/src/main/java/org/talend/sdk/component/design/manager/DesignModel.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) 2006-2017 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.design.manager;
+
+import java.util.Collection;
+
+import lombok.Data;
+
+/**
+ * Stores component meta information required at design time
+ */
+@Data
+public class DesignModel {
+
+    /**
+     * Component unique id
+     */
+    private final String id;
+    
+    /**
+     * Component name
+     */
+    private final String name;
```

Comment:
We dont need the name if we have the id

Reply:
As I understand ComponentManager is to be used for runtime, so we don't want have there things not related in runtime. 
My idea is to move eventually name and icon from BaseMeta to DesignModel as these info is not needed in runtime, but needed in design.

Reply:
we need the name into the runtime since it can be used for lookups (id is really just a workaround to pass it through the URL). +1 to move icons

Reply:
removed name from Model. 
icons to be done